test/fixtures/ol/index.html

```diff
@@ -59,4 +59,13 @@
   <li><p><input type="checkbox">Echo</p></li>
   <li><p><input type="checkbox"><strong>Foxtrot</strong></p></li>
   <li><p><input type="checkbox"> <strong>Golf</strong></p></li>
+  <li>
+    <p>
+      <input type="checkbox"> Hotel
+    </p>
+  </li>
+  <li><input type="checkbox"> India</li>
+  <li>
+    <input type="checkbox"> Juliet
+  </li>
 </ol>
```

Comment:
Maybe a test without anything after it! Particularly without whitespace, like the current new India one

Reply:
Hmmm, that’s an interesting edge case I didn’t think about! What would you expect for output here? The current code creates an empty task list item, e.g:

```js
{
  type: "listItem",
  checked: false,
  children: []
}
```

…which renders like a normal list item, without even a checkbox (adding the Juliet item before it just so it’s clear):

```markdown
* [ ] Juliet
*
```

Is that OK? This is kind of a weird situation (maybe like the things I was thinking there could be warnings about). I’m not sure thare’s any really reasonable result here.

Reply:
Sorry for the wait. Yes, I think that’s fine. I think no checkbox is rendered because a paragraph is required by GFM. And those have to be filled with something? As far as I remember?

Reply:
OK, great, will leave this case functioning as it currently does and get the rest of this done this week. 👍

> I think no checkbox is rendered because a paragraph is required by GFM. And those have to be filled with something? As far as I remember?

Yeah, that’s what’s happening. FWIW, what I was trying to get at was whether you’d want to output a different tree in this case because the above current behavior doesn’t render/outputs an invalid mdast tree. For example, putting in a non-breaking space:

```markdown
* [ ] Juliet
* [ ]  
```

Or a comment:

```markdown
* [ ] Juliet
* [ ] <!-- No text -->
```

Or just backing out and treating it like a checkbox in the middle of text gets treated, rather than as a task list item:

```markdown
* [ ] Juliet
* \[ ]
```

The first two will successfully render as a checklist item when converting back to HTML; the last one does not, but matches behavior elsewhere in this module. There are lots of creative options along these lines.

BUT all of these are doing some kind of funny behavior to preserve the intent of the input. I definitely understand there are reasons you might consider the current garbage-in-garbage-out behavior better or more straightforward.